### DISCUSSION TITLE
GitHub-flavored alerts considered harmful

### DISCUSSION THREAD

**Discussion by meduzen** [vuejs/vitepress#3540] [General]
Hello! With the recent addition of GitHub-flavored alerts, I’ve been rethinking about the value they bring inside and outside of GitHub.

I wrote an article about it, but basically [the issues I raise](https://blog.mehdi.cc/articles/github-alerts-markdown-syntax#the-issues) are exactly the same as the ones already pointed in the [GitHub community discussion](https://github.com/orgs/community/discussions/16925):
- incorrect semantic (leading to [this request](https://github.com/vuejs/vitepress/issues/3512) being posted on Vitepress side, for example);
- heavy limitations (~~1 paragraph~~, no control over the label).

Because of these, they are a sub-par Vitepress container with an additional (and unstable) syntax to learn, and I’m in favor of reconsidering their presence in Vite to avoid encouraging bad semantic practices.

**Edit 2024-02-04**: updated my rant ([summary of the changes](https://blog.mehdi.cc/articles/github-alerts-markdown-syntax#updates-to-this-article)) with new information. Saving you one click:
- @sinsukehlab has made a [lot of tests](https://github.com/sinsukehlab/NOTE-test/issues/1) 😄 that showcase how GitHub alerts render in various situations.
- Vitepress is apparently [adding a setting](https://github.com/vuejs/vitepress/compare/v1.0.0-rc.41...67a9964c4e6ffdb0e644624d075b8dd20f477686) at site config level to disable the GitHub syntax.

**Comment by brc-dd** [comment]
It actually supports block elements, just not changing the title:

> [!NOTE]  
> This is a paragraph
> > This is a blockquote.
> ```ts
> console.log('hello from ts')
> ```

**Reply by sinsukehlab** [reply]
The main point is that alerts violate the spirit of Markdown.
Using `> ` to make a `<div>` section instead of a `<blockquote>` section, fixed label, Anglocentrism, and so on.
Also they can't even make alerts compatible with other elements of GitHub Flavored Markdown.
For example, they've not been able to enable syntax highlighting in alerts for months.
> Ordinary blockquote
> ```python
> print(hello)
> ```

> [!NOTE]
> ```python
> print(hello)
> ```

**Reply by brc-dd** [reply]
If you want to rant about, go rant in their community discussion. I honestly don’t give a fuck and I’m not recommending any particular way. We are just supporting what github supports. The github flavored alerts are for the users who want their markdown files to look good in github previews. If github updates their spec regarding this, we’ll adjust things here. And you’re free to use the non-standard custom blocks instead if you prefer them.

**Reply by pboling** [reply]
I feel like people should rant anywhere they find a pile of dog shit.  If you choose to support the same pile of dog shit in a way that multiplies and proliferates that pile... expect rants.
